configuration/secrets.rst

```diff
@@ -0,0 +1,307 @@
+.. index::
+    single: Secrets
+
+How to Keep Sensitive Informations Secret
+=========================================
+
+In :doc:`/configuration` and :doc:`/configuration/environment_variables`, you
+learned how to manage your application configuration. In this article you'll
+learn how to easily anbd saftly configure your application with sensitive
+information such as credentials and passwords without exposing theme.
+
+.. _secrets-configuration:
+
+Configuration
+-------------
+
+In order to use secrets you have to enable the feature in the framework's
+
+configuration:
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            secrets:
+                # encrypted_secrets_dir: '%kernel.project_dir%/config/secrets/%kernel.environment%/'
+                # encryption_key: '%kernel.project_dir%/config/secrets/encryption_%kernel.environment%.key'
+
+    .. code-block:: xml
+
+            <!-- config/packages/framework.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:framework="http://symfony.com/schema/dic/framework"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd
+                    http://symfony.com/schema/dic/framework https://symfony.com/schema/dic/framework/framework-1.0.xsd"
+            >
+                <framework:config secret="%env(APP_SECRET)%">
+                    <framework:secrets/>
+                </framework:config>
+            </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        $container->loadFromExtension('framework', [
+            'secrets' => true,
+        ]);
+
+.. _secrets-generate-key:
+
+Generate an Encryption Key
+--------------------------
+
+Before creating a new ``secret``, you need to create ann ``encryption key``.
+This can be done with the provided commande ``secrets:generate-key``.
+
+.. code-block:: terminal
+
+    $ php bin/console --env prod secrets:generate-key
+
+This command will generate a new ``encryption key`` in.
+``%kernel.project_dir%/config/secrets/encryption_%kernel.environment%.key``
+
+.. note::
+
+    In order to use Symfony's built-in Secret storage, you will need the sodium
+    PHP extension. Ensure that it is installed.
```

Comment:
maybe add a note or an url referencing this or how to install if not the case?

Reply:
done.